File: LeetCode/Java Solutions/Image Smoother/image.java
public class Solution {
    public int[][] imageSmoother(int[][] img) {
        if (img == null || img.length == 0 || img[0].length == 0) {
            return new int[0][0];
        }

        int rows = img.length;
        int cols = img[0].length;
        int[][] res = new int[rows][cols];

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                int count = 0;
                int sum = 0;

                for (int x = Math.max(0, i - 1); x < Math.min(i + 2, rows); x++) {
                    for (int y = Math.max(0, j - 1); y < Math.min(j + 2, cols); y++) {
                        sum += img[x][y];
                        count++;
                    }
                }

                res[i][j] = sum / count;
            }
        }

        return res;
    }
}

File: LeetCode/Python Solutions/Image Smoother/image.py
from typing import List

class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        if not img:
            return []

        rows, cols = len(img), len(img[0])
        res = [[0 for _ in range(cols)] for _ in range(rows)]

        for i in range(rows):
            for j in range(cols):
                count = 0
                _sum = 0

                for x in range(max(0, i-1), min(i+2, rows)):
                    for y in range(max(0, j-1), min(j+2, cols)):
                        _sum += img[x][y]
                        count += 1

                res[i][j] = _sum // count

        return res
